utils/cv/movenet.py

```diff
@@ -0,0 +1,452 @@
+        img_meta = coco.imgs[img_id]
+        ann_ids = coco.getAnnIds(imgIds=img_id)
+        anns = coco.loadAnns(ann_ids)
+        img_file_name = img_meta['file_name']
+        w = img_meta['width']
+        h = img_meta['height']
+        url = img_meta['coco_url']
+        
+        yield img_id, img_file_name, w, h, url, anns
+
+def convert_to_df(coco, data_set):
+    images_data = []
+    persons_data = []
+    gen = get_meta(coco)
+    for img_id, img_fname, w, h, url, meta in gen:        
+        images_data.append({
+            'image_id': int(img_id),
+            'src_set_image_id': int(img_id), # repeat id to reference after join
+            'coco_url': url,
+            'path': data_set + '/' + img_fname,
+            'width': int(w),
+            'height': int(h)
+        })
+        for m in meta:
+            persons_data.append({
+                'ann_id': m['id'],
+                'image_id': m['image_id'],
+                'is_crowd': m['iscrowd'],
+                'bbox': m['bbox'],
+                'bbox_area' : m['bbox'][2] * m['bbox'][3],
+                'area': m['area'],
+                'num_keypoints': m['num_keypoints'],
+                'keypoints': m['keypoints'],
+                'segmentation': m['segmentation']
+            })
+
+    images_df = pd.DataFrame(images_data)
+    images_df.set_index('image_id', inplace=True)
+
+    persons_df = pd.DataFrame(persons_data)
+    persons_df.set_index('image_id', inplace=True)
+
+    return images_df, persons_df
+
+def get_df(path_to_val_anns):    
+    val_coco = COCO(path_to_val_anns) # load annotations for validation set
+    images_df, persons_df = convert_to_df(val_coco, 'val2017')
+    val_coco_df = pd.merge(images_df, persons_df, right_index=True, left_index=True)
+
+    return val_coco_df
+    # ^ Dataframe containing all val and test keypoint annotations
+
+
+
+def transform_bbox_square(bbox, slack=1):
+    """
+        Transforms a bounding box anchored at top left corner of shape () to a square with
+        edge length being the larger of the bounding box's height or width.
+        Only supports square aspect ratios currently.
+        ## Parameters
+        bbox : {tuple or ndarray of len 4}
+            Given as two points, anchored at top left of image being 0,0: left, upper, right, lower
+        slack : {int, float}
+            The amount of extra padding that should be applied to the edges of the bounding box after
+            transforming
+        ##
+    """
+    x, y, w, h = [i for i in bbox]  # (x,y,w,h) anchored to top left
+    center_x = x+w/2
+    center_y = y+h/2
+
+    if w >= h:
+        new_w = w
+        new_h = w
+    else:
+        new_w = h
+        new_h = h
+
+    new_w *= slack  # add slack to bbox
+    new_h *= slack  # add slack to bbox
+    new_x = center_x - new_w/2
+    new_y = center_y - new_h/2
+    return (round(new_x), round(new_y), round(new_x+new_w), round(new_y+new_h))
+
+class MovenetDataset:
```

Comment:
> Nice PR! Needs few improvements before merging though.
> 
> Please see comments on few details to be polished up. On top of that:
> 
> * setup_deb.sh file has to be updated with new dependencies, like:
>   
> > ModuleNotFoundError: No module named 'tifffile'
> 
> * Read-me suggests run command that fails:
> 
> > python3 run.py -m lite-model_efficientdet_lite2_detection_default_1.tflite -p int8 --framework tf
> > **run.py: error: unrecognized arguments: -p int8**
> 
> * runner should exit gracefully if dataset files' paths were not provided by user, printing what's the issue, instead it throws a bit of a cryptic error:
> 
> > python3 run.py -m lite-model_efficientdet_lite2_detection_default_1.tflite  --framework tf
> > **KeyError: "None of ['image_id'] are in the columns"**
> 
> * model runner should be under computer_vision/pose_estimation directory (not yet existing)


1. setup_deb.sh file is not modified  in my branch. It is modified in main branch.
2. Now the README is updated and the model is movenet and not efficientdet.

Reply:
Yes, this is generic for single-person pose estimation datasets which can be used for any model. Therefore I renamed the class and kept it in utils as it can be used for other pose detection models.